Tools/CISupport/ews-build/steps.py

```diff
@@ -2040,27 +2057,40 @@ def analyzeResults(self):
             compile_without_patch_step = CompileJSCWithoutPatch.name
         compile_without_patch_result = self.getStepResult(compile_without_patch_step)
 
+        patch_id = self.getProperty('patch_id', '')
+        pr_number = self.getProperty('github.number')
+
         if compile_without_patch_result == FAILURE:
-            message = 'Unable to build WebKit without patch, retrying build'
+            if pr_number:
+                message = 'Unable to build WebKit without pull request'
+            else:
+                message = 'Unable to build WebKit without patch, retrying build'
+
             self.descriptionDone = message
             self.send_email_for_preexisting_build_failure()
             self.finished(FAILURE)
-            self.build.buildFinished([message], RETRY)
+            self.build.buildFinished([message], FAILURE if pr_number else RETRY)
             return defer.succeed(None)
 
         self.build.results = FAILURE
-        patch_id = self.getProperty('patch_id', '')
-        message = 'Patch {} does not build'.format(patch_id)
+        sha = self.getProperty('github.head.sha')
+        if sha and pr_number:
+            message = 'Hash {} for PR #{} does not build'.format(sha[:HASH_LENGTH_TO_DISPLAY], pr_number)
```

Comment:
Nit: 'PR #{}' => 'PR {}' to be consistent with other places in this file. If you want to change to include hash that's fine as well, but please do it in other places as well to be consistent.